docs/src/main/asciidoc/deploying-to-google-cloud.adoc

```diff
@@ -0,0 +1,109 @@
+
+This guide will take as input an application developed in the link:datasource.adoc[datasource guide].
+
+Make sure you have the application at hand working locally.
+
+
+== Solution
+
+We recommend to follow the instructions in the next sections and build the application step by step.
+
+== Connecting to CloudSQL
+
+Google CloudSQL managed service allows 4 kinds of connection :
+
+. Using public IP
+. Using private IP
+. Using Cloud SQL Proxy
+. Using service account
+
+The first two don't need anymore details, simply connect to you database following the link:datasource.adoc[datasource guide].
+The third allows you to connect as if locally, but is not available for AppEngine or CloudRun where you only deploy one artefact.
+Please check link:https://cloud.google.com/sql/docs/postgres/external-connection-methods?hl=en[Connection options for external applications].
+
+This guide will help you through the fourth possibility : connecting using service account.
+
+== Adding necessary dependencies to your application.
+
+You will need to add the _Cloud SQL Postgres Socket Factory_ and postgresql driver dependencies.
+
+With maven :
+[source,xml, subs="attributes"]
+----
+<!-- https://mvnrepository.com/artifact/com.google.cloud.sql/postgres-socket-factory -->
+<dependency>
+    <groupId>com.google.cloud.sql</groupId>
+    <artifactId>postgres-socket-factory</artifactId>
+    <version>1.0.15</version>
+</dependency>
+<!-- https://mvnrepository.com/artifact/io.quarkus/quarkus-jdbc-postgresql -->
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-jdbc-postgresql</artifactId>
+</dependency>
+----
+
+With Gradle :
+[source,groovy, subs="attributes"]
+----
+// https://mvnrepository.com/artifact/com.google.cloud.sql/postgres-socket-factory
+compile group: 'com.google.cloud.sql', name: 'postgres-socket-factory', version: '1.0.15'
+// https://mvnrepository.com/artifact/io.quarkus/quarkus-jdbc-postgresql
+compile group: 'io.quarkus', name: 'quarkus-jdbc-postgresql', version: '1.1.1.Final'
+----
+
+== Configuring the CloudSQL instance
+
+If you haven't already please follow the link:https://cloud.google.com/sql/docs/postgres/create-instance[Creating instances guide].
+
+Ensure you have a service account with _Cloud SQL Client_ role (minimum necessary role), and get your instance connection name (`PROJECT_ID:REGION:INSTANCE_ID`).
+
+== Configuring the application
+
+Configure your quarkus application as follows :
+
+[source,properties]
+--
+quarkus.datasource.url=jdbc:postgresql:///${DB_NAME}
+quarkus.datasource.driver=org.postgresql.Driver
+quarkus.datasource.username = ${DB_USER}
+quarkus.datasource.password = ${DB_PASSWORD}
+quarkus.datasource.socket-factory=com.google.cloud.sql.postgres.SocketFactory
+quarkus.datasource.additional-jdbc-properties=cloud-sql-instance=project:zone:db-name
+--
+
+Prefere the use of environment variables that will allow you to connect to all your env depending on your runtime values
```

Comment:
This sentence is a bit confusing

Reply:
I changed this sentence. Please take a look now if it is more clear